client/settings/advanced-settings/client-secret-encryption-toggle.js

```diff
@@ -0,0 +1,47 @@
+/**
+ * External dependencies
+ */
+import { CheckboxControl } from '@wordpress/components';
+import { __ } from '@wordpress/i18n';
+import { useEffect, useRef } from '@wordpress/element';
+
+/**
+ * Internal dependencies
+ */
+import { useClientSecretEncryption } from 'wcpay/data';
+
+const ClientSecretEncryptionToggle = () => {
+	const [
+		isClientSecretEncryptionEnabled,
+		updateIsClientSecretEncryptionEnabled,
+	] = useClientSecretEncryption();
+
+	const headingRef = useRef( null );
+
+	useEffect( () => {
+		if ( ! headingRef.current ) return;
+
+		headingRef.current.focus();
+	}, [] );
+
+	const handleClientSecretEncryptionStatusChange = ( value ) => {
+		updateIsClientSecretEncryptionEnabled( value );
+	};
+
+	return (
+		<CheckboxControl
+			label={ __(
+				'Enable UPE Public Key Encryption',
```

Comment:
Since this feature is available on non-UPE checkout we should probably remove 'UPE' from the label string. What do you think?

Reply:
As we discussed briefly earlier, since the client secret is only exposed to the client when UPE is enabled, we should  hide this checkbox feature on the Payments > Settings page if UPE is not enabled.

Reply:
Instead hiding the checkbox, I removed the `UPE` word from the label on [4cd98a9](https://github.com/Automattic/woocommerce-payments/pull/4782/commits/4cd98a95bb03708d267e393e5de4bf1a1383e2f3).